### PR TITLE
Add a test for shadow stack allocation before an external call.

### PR DESCRIPTION
**Pull request by vext01**
An external call can "call-back" to interpreter code, so we still need to allocate shadow space.

Requires https://github.com/ykjit/ykllvm/pull/217

### CODE REVIEW

**Comment by vext01** [comment]
hold off on this for this and the companion for a moment actually.

File: tests/src/lib.rs [hunk 1]
```diff
@@ -21,6 +21,7 @@ pub static EXTRA_LINK: LazyLock<HashMap<&'static str, Vec<ExtraLinkage>>> = Lazy
         "loopy_funcs_not_inlined_by_default.c",
         "not_loopy_funcs_inlined_by_default.c",
         "reentrant.c",
+        "shadow_reentrant.c",
```
Comment (ltratt) [inline tests/src/lib.rs:24]: I don't immediately see why this test needs extra linking?

Comment (vext01) [reply tests/src/lib.rs:24]: It calls to an external function `call_callback()` which is compiled externally.

Comment (ltratt) [reply tests/src/lib.rs:24]: Ah!

**Comment by vext01** [comment]
Force pushed test fixes that I didn't anticipate. Go ahead.

**Comment by vext01** [comment]
Oh this will need a ykllvm sync once the compiler PR is merged.

**Comment by ltratt** [comment]
Please force push when that's ready.

**Comment by vext01** [comment]
synced and ready.

(I note that the ykllvm sync also pulled in some SWT-related stuff)

**Comment by ltratt** [comment]
Yes the SWT parts are expected.

**Comment by vext01** [comment]
Ready when you are then.

**Comment by ltratt** [comment]
```
11:58:56 ***** FILE 'events.lua'*****
11:58:56 testing metatables
11:58:56 ..Segmentation fault (core dumped)
11:58:59 + status=139
11:58:59 + [ 1 -eq 1 ]
```

Do we think that's a non-deterministic failure, or perhaps evidence of a problem caused the ykllvm change?

**Comment by vext01** [comment]
Hrm, hard to say. I have a try_repeat going to see if I can repro it locally.

Maybe retry it?

**Comment by vext01** [comment]
My run of 100 serialised debug runs just finished without hitting this seg :thinking:

**Comment by ltratt** [comment]
OK, well let's retry, and see what happens.

**Comment by ltratt** [comment]
OK, it's crashed in the same place, so I think we can assume we've hit an actual problem.

**Comment by vext01** [comment]
I've pushed an assertion that may save us some debugging effort in the future.

With https://github.com/ykjit/ykllvm/pull/219 I think this should merge (if the benchmarks turn out to be ok).

**Comment by vext01** [comment]
This will need an llvm sync after https://github.com/ykjit/ykllvm/pull/219 is in.

**Comment by vext01** [comment]
synced llvm.

**Pull request merged**